File: SwiftySheetsDemo/Parser/AST/ASTStringLiteral.m
//
//  ASTStringLiteral.m
//  SwiftySheetsDemo
//
//  Created by imihaly on 09/07/16.
//  Copyright © 2016 Agnes Vasarhelyi. All rights reserved.
//

#import "ASTStringLiteral.h"

@interface ASTStringLiteral()
@property(nonatomic, copy, readwrite) NSString *value;
@end

@implementation ASTStringLiteral

+ (instancetype)literalWithString:(NSString *)string {
	return [[self alloc] initWithString:string];
}

- (instancetype)initWithString:(NSString *)string {
	self = [super init];
	if(self) {
		self.value = string;
	}
	return self;
}

- (ASTExprType)type {
	return ASTExprType_String;
}

- (id)eval {
	return self.value;
}

- (NSString *)description {
	return [NSString stringWithFormat:@"Str<%@>", self.value];
}

@end

File: SwiftySheetsDemo/Parser/AST/ASTStringLiteral.swift
//
//  ASTStringLiteral.swift
//  SwiftySheetsDemo
//
//  Created by Alföldi Norbert on 19/07/16.
//  Copyright © 2016 Agnes Vasarhelyi. All rights reserved.
//

import Foundation

@objc public class ASTStringLiteral : NSObject, ASTExpression, StringLiteralConvertible {
	public var value: String
	
	public required init(stringLiteral value: StringLiteralType) {
		self.value = value
	}
	
	public typealias UnicodeScalarLiteralType = StringLiteralType
	public required init(unicodeScalarLiteral value: UnicodeScalarLiteralType) {
		self.value = value
	}
	
	public typealias ExtendedGraphemeClusterLiteralType = StringLiteralType
	public required init(extendedGraphemeClusterLiteral value: ExtendedGraphemeClusterLiteralType) {
		self.value = value
	}
	
	public func eval() -> AnyObject {
		return self.value
	}
}
